server/src/main/java/io/crate/expression/scalar/timestamp/CurrentTimeFunction.java

```diff
@@ -0,0 +1,100 @@
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar.timestamp;
+
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+import io.crate.types.TimeTZ;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Locale;
+
+public class CurrentTimeFunction extends Scalar<TimeTZ, Integer> {
+
+    public static final String NAME = "current_time";
+    private static final int MICRO_PRECISION = 6; // microseconds
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.TIMETZ.getTypeSignature()
+            ),
+            CurrentTimeFunction::new
+        );
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.TIMETZ.getTypeSignature()
+            ),
+            CurrentTimeFunction::new
+        );
+    }
+
+    private final Signature signature;
+    private final Signature boundSignature;
+
+    private CurrentTimeFunction(Signature signature, Signature boundSignature) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+    }
+
+    @Override
+    @SafeVarargs
+    public final TimeTZ evaluate(TransactionContext txnCtx, Input<Integer>... args) {
+        Integer precision = MICRO_PRECISION;
+        if (args.length == 1) {
+            precision = args[0].value();
+            if (precision == null) {
+                throw new IllegalArgumentException(String.format(
+                    Locale.ENGLISH,
+                    "NULL precision not supported for %s",
+                    signature.getName().name()));
+            }
+        }
+        if (precision < 0 || precision > MICRO_PRECISION) {
```

Comment:
This could move into the previous `if` block. We know that the precision is valid if it comes from the default value if there are 0 arguments.